src/pifile.c

```diff
@@ -42,6 +37,13 @@ static void *__pifile_open(char *name, int flags)
         return NULL;
     }
 
+    if ((base < 0x08000000) || (base >= 0x09000000)) {
+        if (flags != O_RDONLY) {
+            errno = EACCES;
+            return NULL;
+        }
+    }
+
```

Comment:
I don't think we need this at this level. In general any device can expose any writeable area at any PI address, so I wouldn't hardcode the range. Eg: many flashcarts allow to write the internal SDRAM by writing at 0x10000000, maybe after enabling writes with some register.

So I would just remove the check and let the user know what they are doing.